dvc/repo/experiments/show.py

```diff
@@ -45,20 +45,21 @@ def _collect_experiment_commit(
             res["params"] = params
 
         res["deps"] = {
-            dep.def_path: {
+            relpath(dep.fs_path, repo.root_dir): {
                 "hash": dep.hash_info.value,
                 "size": dep.meta.size,
                 "nfiles": dep.meta.nfiles,
             }
             for dep in repo.index.deps
             if not isinstance(dep, (ParamsDependency, RepoDependency))
         }
-
         res["outs"] = {
-            out.def_path: {
+            relpath(out.fs_path, repo.root_dir): {
                 "hash": out.hash_info.value,
                 "size": out.meta.size,
                 "nfiles": out.meta.nfiles,
+                "use_cache": out.use_cache,
+                "is_data_source": out.stage.is_data_source,
```

Comment:
[Q] Do I have everything I need to work this out:

> ```
> for dep in res[“deps”].values():
>     for out in res[“outs”].values():
>         if dep[“fs_path”] == out[“fs_path”]:
>             # THIS IS AN INTERMEDIATE ARTIFACT
>             dep["intermediate"] = True
>             out["intermediate"] = True
>             if out[“use_cache”]:
>                 # DEP IS DVC TRACKED
>                 dep["use_cache"] = True
>             else:
>                 # DEP IS GIT-TRACKED 
>                 dep["use_cache"] = False
> ```

The answer is yes

> Given that the rules about deciding what files should be included appear to be currently unclear and product-dependent I would prefer not to include this logic in DVC.

This does make sense to me. However, there will be much less chance of providing consistency across dependencies if we push this logic out to the consumers.

Reply:
@mattseddon where does this snippet come from?

I see that `is_data_source` is included which is what we need I believe, @daavoo could you remind please what is the definition of the `out.stage.is_data_source`?

Reply:
https://github.com/iterative/dvc/issues/7575#issuecomment-1116456065

Reply:
Correct code is in the description?